lib/Parser/Parser.yy

```diff
@@ -87,118 +87,453 @@
+    ;
+
+function_body:
+      block
+    | semi
+    ;
+
+block:
+      lBrace statement_list rBrace
+    ;
+
+statement_list:
+      /* empty */
+    | statement_list statement
+    ;
+
+statement:
+      block
+    | expression semi
+    | var_stmt
+    | let_stmt
+    | return_stmt
     | if_stmt
     | while_stmt
-    | return_stmt
+    | for_stmt
+    | break_stmt
+    | continue_stmt
+    | assignment_stmt
+    | semi
     ;
 
-expr_stmt
-    : expr semi
+var_stmt:
+      varKw ident type_opt equal_opt expression_opt semi
+    {
+        std::cerr << "Parsed var statement" << std::endl;
+    }
+    ;
+
+type_opt:
+      /* empty */
+    | colon type
     ;
 
-if_stmt
-    : ifKw lParen expr rParen statement
-    | ifKw lParen expr rParen statement elseKw statement
+equal_opt:
+      /* empty */
+    | equal
     ;
 
-while_stmt
-    : whileKw lParen expr rParen statement
+expression_opt:
+      /* empty */
+    | expression
     ;
 
-return_stmt
-    : returnKw semi
-    | returnKw expr semi
+let_stmt:
+      letKw ident type_opt equal expression semi
+    {
+        std::cerr << "Parsed let statement" << std::endl;
+    }
     ;
 
-expr
-    : ident
+return_stmt:
+      returnKw expression_opt semi
+    {
+        std::cerr << "Parsed return statement" << std::endl;
+    }
+    ;
+
+if_stmt:
+      ifKw expression block else_opt
+    {
+        std::cerr << "Parsed if statement" << std::endl;
+    }
+    ;
+
+else_opt:
+      /* empty */
+    | elseKw statement
+    ;
+
+while_stmt:
+      whileKw expression block
+    {
+        std::cerr << "Parsed while statement" << std::endl;
+    }
+    ;
+
+for_stmt:
+      forKw ident inKw expression block
+    {
+        std::cerr << "Parsed for statement" << std::endl;
+    }
+    ;
+
+break_stmt:
+      breakKw semi
+    {
+        std::cerr << "Parsed break statement" << std::endl;
+    }
+    ;
+
+continue_stmt:
+      continueKw semi
+    {
+        std::cerr << "Parsed continue statement" << std::endl;
+    }
+    ;
+
+assignment_stmt:
+      expression equal expression semi
+    {
+        std::cerr << "Parsed assignment statement" << std::endl;
+    }
+    ;
+
+expression:
+      literal
+    | ident
+    | function_call
+    | binary_expression
+    | unary_expression
+    | paren_expression
+    | initializer_list
+    | ternary_expression
+    | field_access
+    | subscript
+    | cast_expression
+    ;
+
+function_call:
+      namespaced_identifier template_arguments_opt lParen rParen
+    | namespaced_identifier template_arguments_opt lParen argument_list rParen
+    ;
+
+template_arguments_opt:
+      /* empty */
+    | template_arguments
+    ;
+
+template_arguments:
+      ltOp type_list gtOp
+    ;
+
+type_list:
+      type
+    | type_list comma type
+    ;
+
+argument_list_opt:
+      /* empty */
+    | argument_list
+    ;
+
+argument_list:
+      expression
+    | argument_list comma expression
+    ;
+
+binary_expression:
+      expression binary_operator expression
+    ;
```

Comment:
this doesn't handle precedence, there should be one rule per precedence group (sorry, the formal grammar should've included this)